### PR TITLE
PIP-1636-tests-with-caper

### PR DESCRIPTION
**Pull request by ottojolanki**
Run tests with caper instead of directly interacting with cromwell. Add the readiness to run pytest workflow.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,31 +17,73 @@ python_defaults: &python_defaults
...
+            echo "export TAG=encodedcc/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_BRANCH}_${CIRCLE_WORKFLOW_ID}" >> ${BASH_ENV}
+            echo "export CROMWELL=./cromwell-70.jar" >> ${BASH_ENV}
+            echo "export WOMTOOL=./womtool-70.jar" >> ${BASH_ENV}
+
+  install_python_requirements:
+    description: "Install pytest workflow requirements"
+    steps:
+      - run: pyenv install 3.7.8
+      - run: pyenv global 3.7.8
+      - run: pip install --upgrade pip
```
Comment (paul-sud) [inline .circleci/config.yml]: Maybe worth pinning pip at least to a major version to avoid breakages.

File: .circleci/config.yml [hunk 2]
```diff
@@ -17,31 +17,73 @@ python_defaults: &python_defaults
...
+    description: "Get machine ready to run caper and pytest-workflow"
+    steps:
+      - set_env_variables
+      - install_python_requirements
+      - download_file:
+          file_url: "https://github.com/broadinstitute/cromwell/releases/download/70/cromwell-70.jar"
+      - download_file:
+          file_url: "https://github.com/broadinstitute/cromwell/releases/download/70/womtool-70.jar"
+
+  run_test_tag:
```
Comment (paul-sud) [inline .circleci/config.yml:64]: I suppose you are planning on upgrading the test suite later? I don't see this used.

File: .circleci/config.yml [hunk 3]
```diff
@@ -17,31 +17,73 @@ python_defaults: &python_defaults
...
+    description: "Run tagged test"
+    parameters:
+      tag:
+        type: string
+      wt:
+        type: string
+    steps:
+      - run:
+          command: pytest -v --tag <<parameters.tag>> --wt <<parameters.wt>> --symlink --kwd
+          no_output_timeout: 60m
```
Comment (paul-sud) [inline .circleci/config.yml]: This seems long for a `no_output_timeout`, but maybe it really is needed. I'm not sure how long the RNA tests usually take.

File: .circleci/config.yml [hunk 4]
```diff
@@ -17,31 +17,73 @@ python_defaults: &python_defaults
...
+            echo "export CROMWELL=./cromwell-70.jar" >> ${BASH_ENV}
+            echo "export WOMTOOL=./womtool-70.jar" >> ${BASH_ENV}
+
+  install_python_requirements:
+    description: "Install pytest workflow requirements"
+    steps:
+      - run: pyenv install 3.7.8
+      - run: pyenv global 3.7.8
+      - run: pip install --upgrade pip
+      - run: pip install -r test_env_requirements.txt
```
Comment (paul-sud) [inline .circleci/config.yml]: I suggest calling this file `requirements-dev.txt` which is more standard.

**Review by paul-sud — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (ottojolanki) [reply .circleci/config.yml]: k. this is copypasta from dnase. 15min is probably more appropriate here.

**Review by paul-sud — APPROVED** [review]

**Pull request merged**